disnake/emoji.py

```diff
@@ -151,16 +159,34 @@ def roles(self) -> List[Role]:
         and count towards a separate limit of 25 emojis.
         """
         guild = self.guild
-        if guild is None:  # pyright: ignore[reportUnnecessaryComparison]
+        if guild is None:
             return []
 
         return [role for role in guild.roles if self._roles.has(role.id)]
 
     @property
-    def guild(self) -> Guild:
-        """:class:`Guild`: The guild this emoji belongs to."""
+    def guild(self) -> Optional[Guild]:
+        """Optional[:class:`Guild`]: The guild this emoji belongs to.
+
+        ``None`` if this is an app emoji.
+
+        .. versionchanged:: 2.10
+
+            This can now return ``None`` if the emoji is an
+            application owned emoji.
+        """
```

Comment:
```suggestion
        """Optional[:class:`Guild`]: The guild this emoji belongs to. ``None`` if this is an app emoji.

        .. versionchanged:: 2.10

            This can now return ``None`` if the emoji is an
            application owned emoji.
        """
```